File: API/picpayclone/src/main/java/com/picpayclone/model/Transaction.java
package com.picpayclone.model;

import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
@Entity
@Table(name = "TRANSACTIONS")
public class Transaction extends BaseEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    @Column(name = "TR_CODE", nullable = false)
    private String code;

    @ManyToOne(cascade = { CascadeType.MERGE })
    @JoinColumn(name = "TR_USER_ORIGIN", nullable = false)
    private User origin;

    @ManyToOne(cascade = { CascadeType.MERGE }, fetch = FetchType.EAGER)
    @JoinColumn(name = "TR_USER_DESTINATION", nullable = false)
    private User destination;

    @Column(name = "TR_DATE_TIME", nullable = false)
    private LocalDateTime dateTime;

    @Column(name = "TR_VALUE", nullable = false)
    private Double value;

}

File: Android/app/src/main/java/com/app/picpayapp/data/transaction/Transaction.kt
package com.app.picpayapp.data.transaction

import com.app.picpayapp.data.CreditCard
import com.app.picpayapp.data.User
import java.util.*

data class Transaction(
    val code: String = "",
    val origin: User = User(),
    val destination: User = User(),
    val dateTime: String = "",
    val isCreditCard: Boolean = false,
    val value: Double = 0.0,
    val creditCard: CreditCard = CreditCard()
) {
    companion object {
        fun hashGenerator(): String = UUID.randomUUID().toString()
    }
}
